extensions/kafka-streams/deployment/src/main/java/io/quarkus/kafka/streams/deployment/KafkaStreamsProcessor.java

```diff
@@ -195,6 +196,18 @@ AdditionalBeanBuildItem registerBean() {
         return AdditionalBeanBuildItem.unremovableOf(KafkaStreamsTopologyManager.class);
     }
 
+    @BuildStep
+    HealthBuildItem addKafkaStreamsTopicsHealthCheck(KafkaStreamsBuildTimeConfig buildTimeConfig) {
+        return new HealthBuildItem("io.quarkus.kafka.streams.runtime.health.KafkaStreamsTopicsHealthCheck",
+                buildTimeConfig.healthEnabled, "kafka-streams");
+    }
+
+    @BuildStep
+    HealthBuildItem addKafkaStreamsStateHealthCheck(KafkaStreamsBuildTimeConfig buildTimeConfig) {
+        return new HealthBuildItem("io.quarkus.kafka.streams.runtime.health.KafkaStreamsStateHealthCheck",
+                buildTimeConfig.healthEnabled, "kafka-streams");
+    }
+
```

Comment:
You can produce those two build items in the same method using a `BuildItemProducer` (ping me if you need some example). Two methods seems overkill for me

Reply:
For my personal knowledge, why is it better to use the `BuildProducer` rather than 2 `BuildStep` ?

Reply:
It's eaisier to read, you will have a single method with the responsibility to produce all build steps related to health.
But as always, it's questionnable ... so do as you want :)